cve_bin_tool/cli.py

```diff
@@ -55,6 +58,11 @@ class Scanner(object):
     def __init__(self, cvedb, checkers=None, logger=None):
         if logger is None:
             logger = LOGGER.getChild(self.__class__.__name__)
+        # update egg before performing any operation
```

Comment:
This should only be done if the package is installed in development mode. We need either a way to check if the production version of the package is installed, or to check if its installed in development mode and only do this if it is

References: https://intel.github.io/dffml/master/contributing/codebase.html#checking-for-development-version

Reply:
we can use an environment variable for that

Reply:
I have checked reference but i didn't get it.

Reply:
Hmmm, the issue here is that developers don't remember to run `bdist_egg`, in which case they probably won't remember to set an environment variable either. Also, what happened to `redirect_stdout`?

Reply:
```python
any(
        list(
            map(
                os.path.isfile,
                list(
                    map(
                        lambda syspath: os.path.join(
                            syspath, "cve-bin-tool.egg-link"
                        ),
                        sys.path,
                    )
                ),
            )
        )
    )
```

Returns true if `cve-bin-tool` was installed with `pip install -e`

Reply:
redirect_stdout has global side effect. so, i am using good old trick of redirecting stdout manually.

Reply:
I added  a function that checks for production build